src/data/sum/sum3.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.fintype.basic
+
+/-!
+# Ternary sum of types
+
+This file defines `sum₃`, a shorthand for the ternary sum of types. In almost all cases,
+`α ⊕ β ⊕ γ`, a nested binary sum, should be preferred over `sum₃ α β γ`. The only point of `sum₃`
+is to offer better pattern-matching properties.
```

Comment:
What are these properties?

Reply:
I should really stop forgetting about `pattern` 🤦

It just works, except that I lost the ability to pass on exactly the instances I was interested in, so a bunch of decidability issues popped up. But that's fixed now.